data/pathfinder/paizo/roleplaying_game/core_rulebook/cr_abilities_race.lst

```diff
@@ -4,163 +4,171 @@ SOURCELONG:Core Rulebook	SOURCESHORT:CR	SOURCEWEB:http://paizo.com/store/downloa
+Languages						KEY:Gnome ~ Languages			CATEGORY:Special Ability	TYPE:RacialTraits.Gnome Racial Trait.Gnome Racial Default.SpecialQuality.Racial Language				VISIBLE:DISPLAY	!PREFACT:1,ABILITIES,Gnome_ReplaceLanguages=true																																		DESC:Gnomes begin play speaking Common, Gnome, and Sylvan. Gnomes with high Intelligence scores can choose from the following: Draconic, Dwarven, Elven, Giant, Goblin, and Orc.																																														TEMPLATE:Race Bonus Language ~ Draconic|Race Bonus Language ~ Dwarven|Race Bonus Language ~ Elven|Race Bonus Language ~ Giant|Race Bonus Language ~ Goblin|Race Bonus Language ~ Orc	AUTO:LANG|Common|Gnome|Sylvan																																																												SOURCEPAGE:p.23
+
+###Block: Obsessive
+# Ability Name			Category of Ability	Type						Description												Stackable?	Multiple?	Choose				Bonus to skill
+Obsessive Craft Bonus		CATEGORY:Internal		TYPE:Gnome Obsessive Skill Bonus	DESC:Gnomes receive a +2 racial bonus on a Craft skill of their choice.		STACK:NO	MULT:YES	CHOOSE:SKILL|TYPE=Craft		BONUS:SKILL|LIST|2|TYPE=Racial
+Obsessive Profession Bonus	CATEGORY:Internal		TYPE:Gnome Obsessive Skill Bonus	DESC:Gnomes receive a +2 racial bonus on a Profession skill of their choice.	STACK:NO	MULT:YES	CHOOSE:SKILL|TYPE=Profession	BONUS:SKILL|LIST|2|TYPE=Racial
+
+
+# ------------------------------
+# Half-Elf
+# ------------------------------
+
+###Block: Racial Abilities ~ grantor/removal MODs in cr_abilities_globalvar.lst
+# Ability Name			Category of Ability
+Racial Traits ~ Half-Elf	CATEGORY:Internal
+
+###Block: Default Racial Traits
+# Ability Name		Unique Key					Category of Ability		Type																	Visible		Prohibited Fact								Define				Description																												Template					Auto Language		Ability								Bonus Ability Pool					Bonus to skill							Modify VAR								Vision			Source Page		Aspects
++2 to One Ability Score	KEY:Half-Elf ~ Ability Scores		CATEGORY:Special Ability	TYPE:RacialTraits.Half-Elf Racial Trait.Half-Elf Racial Default.SpecialQuality.Racial Ability Scores	VISIBLE:DISPLAY	!PREFACT:1,ABILITIES,HalfElf_ReplaceAbilityScores=true						DESC:Half-elf characters get a +2 bonus to one ability score of their choice at creation to represent their varied nature.																												BONUS:ABILITYPOOL|Ability Bonus|1																									SOURCEPAGE:p.24
+Medium			KEY:Half-Elf ~ Size			CATEGORY:Special Ability	TYPE:RacialTraits.Half-Elf Racial Trait.Half-Elf Racial Default.Sp
```

Comment:
This right here breaks the Halfling Race output. Part 1.

Reply:
It seems to have selected the wrong line... Halfling Saves. The BONUS tag MUST be in the race file. Otherwise the output does NOT detect it as RACE.

@mertonmonk might consider fixing this.

Reply:
It ate my earlier comment.
I have reverted the change as requested, although that goes against the design philosophy I was following. The code needs to be fixed so that TYPE bonuses of different file types stack properly. The reason there was an error here is because TYPE=Racial in an ability file is treated different from TYPE=Racial in a races file. Although, racial bonuses aren't even referred to after racial abilities, and except for the particular sheet that the tests use, the racial bonus is only counted as part of the misc bonus on most output. If a fix is not forthcoming, it might be more prudent to fix the test to look for abilities that add to the TYPE=Racial bonus as well as the normal race bonus output.